sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc

```diff
@@ -0,0 +1,413 @@
+_Effects_: Loads `ElementsPerWorkItem` elements from `in_ptr` to `out`
+using the `gh` group helper object.
+Properties may specify xref:data_placement[data placement].
+Default placement is a blocked one:
+`out[i]` = `in_ptr[gh.get_local_linear_id() * ElementsPerWorkItem + i];`
+in striped case:
+`out[i]` = `in_ptr[gh.get_local_linear_id() + gh.get_local_linear_range() * i];`
+for `i` between `0` and `ElementsPerWorkItem`.
+
+.(7) _Preconditions_:
+* `out_ptr` must be the same for all work-items in the group.
+* `in` must be a `sycl::span` made from a pointer to the private memory space.
+
+._Constraints_:
+* Only available if `GroupHelper` is a
+work-group or sub-group or `GroupHelper` was created with a
+work-group or sub-group and some associated scratch space
+(link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[`group_with_scratchpad`]).
+* `OutputIteratorT` suppose to be a random access iterator.
+* InputT must be convertible to value type of `OutputIteratorT`.
+
+_Effects_: Stores `ElementsPerWorkItem` elements from `in` span to `out_ptr`
+using the `gh` group helper object.
+
+Properties may specify xref:data_placement[data placement].
+Default placement is a blocked one:
+`out_ptr[gh.get_local_linear_id() * ItemsPerWorkItem + i]` = `in[i];`
+in striped case:
+`out_ptr[gh.get_local_linear_id() + gh.get_local_linear_range() * i]` = `in[i];`
+for `i` between `0` and `ItemsPerWorkItem`.
+
+(8)_Effects_: Returns size of temporary memory (in bytes) that is required for
+scratch space in `GroupHelper`. Result depends on type `T`, `ElementsPerWorkItem`
+and the scope parameter: use sycl::memory_scope::work_group to get memory size
+required for each work-group; use sycl::memory_scope::sub_group to get memory
+size required for each sub-group. If other scope values are passed, behavior is
+unspecified.
+
+== Data placement
+
+anchor:data_placement[]
+
+To specify a correct data placement for placing of resulting data
+there is a enum (proposed in link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[`group_sort extention`]):
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Properties:
+enum class group_algorithm_data_placement {
+  blocked,
+  striped
+};
+
+}
+----
+
+Data placement layout example on group_load:
+
+* ElementsPerWorkItem = 4
+* 3 work-items in the group
+* input is: in_ptr[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11}
+
+Consider 2 layouts:
+
+1.`sycl::ext::oneapi::experimental::group_algorithm_data_placement::blocked`.
+
+|===
+|Work-item id|Output stored in private fixed-size array
+
+|0
+|{0, 1, 2, 3}
+|1
+|{4, 5, 6, 7}
+|2
+|{8, 9, 10, 11}
+|===
+
+2.`sycl::ext::oneapi::experimental::group_algorithm_data_placement::striped`.
+
+|===
+|Work-item id|Output stored in private fixed-size array
+
+|0
+|{0, 3, 6, 9}
+|1
+|{1, 4, 7, 10}
+|2
+|{2, 5, 8, 11}
+|===
+
+There is a property that satisfies
```

Comment:
I'm not a native speaker, so might be wrong here. That said, my first impression was that one of other extensions has already defined that property, but it seems not to be so (because group sort has separate input/output properties for that, albeit using the same wording as here). 

Can we change it to something along the lines of "this extension adds new property(ies)..."?

Reply:
Good point, changed description and added references to group sort's input/output_data_placement